service/api_task.go

```diff
@@ -608,25 +619,35 @@ func (as *APIServer) NextTask(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 		if aliasQueue != nil {
-			nextTask, err = assignNextAvailableTask(ctx, aliasQueue, as.taskAliasDispatcher, h)
+			nextTask, shouldRunTeardown, err = assignNextAvailableTask(ctx, aliasQueue, as.taskAliasDispatcher, h, details)
```

Comment:
I think this means that task groups don't work with distro aliases, but I think that should be out of the scope of this PR. Could you open a ticket for this? The problem is that if a host gets a task group task in its alias queue, it may not get the next task from that alias queue the next time it runs.

Reply:
Ah, do you mean that pre these changes task groups also didn't work with distro aliases? 

Would it work to still check the alias queue if nextTask is nil (and not also check `shouldRunTeardown` line 613)? And only teardown if shouldRunTeardown in the regular queue AND (alias queue is nil OR shouldRunTeardown from alias queue)? So that way we still consider if the task group was from the alias queue? 
(Not actually sure how alias queues work and if there's some other nuance I'm missing.)

Reply:
I'm saying that this could happen:
1. Host asks for task. There's no task in its distro queue, so we get a task from an alias queue.
2. The host finishes the task and asks for another one. In the meantime, tasks have entered the distro queue.
3. We iterate over the distro queue, and, having not found a task in the host's current task group, we return a different task.

But I think this is unrelated to your change. Also, no one is using aliases yet. So I think we should just open a ticket for the problem.

Reply:
I think the quick fix is that we just block task groups from aliases for the moment until we have a way around it.

Reply:
Right, but I think that should be a separate ticket.

Reply:
Done https://jira.mongodb.org/browse/EVG-6692